students/becky_larson/lesson09/donor_models.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env python3
+
+"""
+Becky Larson
+Created 10/5/2020
+Updated 10/11/2020
+"""
+
+"""
+Donor Class: responsible for donor data encapsulation
+
+This class holds information about a single donor.
+Donor has attributes, properties ^ methods to provide access to donor.
+Only code accesses info about a single donor should be part of this class.
+"""
+
+
+class Donor:
+    def __init__(self, donor_name="", donations=None):
+        """ Initialize the Donor"""
+        """
+        :donor_name:   Name of donor
+        :donations:    Donations the donor has made
+        """
+
+        self._donor_name = donor_name
+        self._donations = []
+
+        if not donations:
+            self._donations = []
+        elif hasattr(donations, "__iter__"):
+            try:
+                [self.add_donation(donation) for donation in donations]
```

Comment:
this is considered an anti-pattern - you'll want to use a regular for loop here.

the above list comprehension does not consume the result from a function (which is None)